### PR TITLE
Hooks refactors 1

### PR DESCRIPTION
**Pull request by truemiller**
## Proposed changes

Refactor hooks to support updated contexts
About half way through ~

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: frontend/hooks/useBalanceContext.ts [hunk 1]
```diff
@@ -39,7 +45,7 @@ export const useServiceBalances = (serviceConfigId: string) => {
   const serviceSafeBalances = useMemo<WalletBalanceResult[]>(
     () =>
       walletBalances?.filter((balance) =>
-        flatAddresses.includes(balance.walletAddress),
+        masterSafes.find(({address}) => balance.walletAddress === address),
```
Comment (Tanya-atatakai) [inline frontend/hooks/useBalanceContext.ts:48]: why to change, it seems masterSafes are in flatAddresses? 
image

File: frontend/hooks/useBalanceContext.ts [hunk 2]
```diff
@@ -52,3 +58,54 @@ export const useServiceBalances = (serviceConfigId: string) => {
...
+
+  const isLowBalance = useMemo(
+    () => masterLowBalances?.length > 0,
+    [masterLowBalances],
+  );
+
+  const masterSafeBalances = useMemo<WalletBalanceResult[]>(
+    () =>
+      walletBalances?.filter((balance) =>
+        masterSafes.find(({address}) => balance.walletAddress === address),
```
Comment (Tanya-atatakai) [inline frontend/hooks/useBalanceContext.ts:99]: I see the TODO, so can ignore, just a thought - does it make sense to use flatAddresses here for consistency?

File: frontend/hooks/useNeedsFunds.ts [hunk 3]
```diff
@@ -1,54 +1,114 @@
...
-      serviceTemplate.configurations[CHAIN_CONFIG.OPTIMISM.chainId]
-        .monthly_gas_estimate;
-    const monthlyGasEstimate = Number(formatUnits(`${gasEstimate}`, 18));
-    const minimumStakedAmountRequired =
-      getMinimumStakedAmountRequired(serviceTemplate);
-
-    return { eth: monthlyGasEstimate, olas: minimumStakedAmountRequired };
+    const results: {
+      [chainId: number]: {
+        [tokenSymbol: string]: number;
```
Comment (Tanya-atatakai) [inline frontend/hooks/useNeedsFunds.ts:40]: something with your linter? 
image

File: frontend/hooks/useNeedsFunds.ts [hunk 4]
```diff
@@ -1,54 +1,114 @@
...
+    );
+
+    return results;
   }, [serviceTemplate]);
 
   const hasEnoughEthForInitialFunding = useMemo(
-    () => (safeBalance?.ETH || 0) >= (serviceFundRequirements?.eth || 0),
-    [serviceFundRequirements?.eth, safeBalance],
+    () => {
+      if (!serviceFundRequirements) return ;
```
Comment (Tanya-atatakai) [inline frontend/hooks/useNeedsFunds.ts:76]: also small, but I noticed in `needsInitialFunding` you return false in similar cases, but here undefined;
for type safety and consistency, maybe you could specify the return type? might help prevent surprises
```
const hasEnoughEthForInitialFunding: boolean = useMemo(
```

**Review by Tanya-atatakai — APPROVED** [review]

File: frontend/hooks/useMultisig.ts [hunk 5]
```diff
@@ -34,3 +35,83 @@ export const useMultisig = (safe: Safe) => {
...
+        );
+        if (safesOnChainId.length === 0) {
+          continue;
+        }
+
+        results[+chainId] = safesOnChainId.map((safe) => ({
+          safeAddress: safe.address,
+          contractCall: new MulticallContract(
+            safe.address,
+            GNOSIS_SAFE_ABI,
```
[the remark this answers is not in the document]
Comment (truemiller) [reply frontend/hooks/useMultisig.ts:76]: the contracts don't have an assigned in chainId in the multicall package, they're processed via chain specific provider in step 2

File: frontend/hooks/useBalanceContext.ts (hunk 1 above)
Comment (truemiller) [reply frontend/hooks/useBalanceContext.ts:48]: hmm.. you're correct this is wrong :|

File: frontend/hooks/useBalanceContext.ts [hunk 6]
```diff
@@ -52,3 +58,54 @@ export const useServiceBalances = (serviceConfigId: string) => {
...
+  const masterLowBalances = useMemo(
+    () => lowBalances?.filter((balance) => balance.walletAddress),
+    [lowBalances],
+  );
+
+  const isLowBalance = useMemo(
+    () => masterLowBalances?.length > 0,
+    [masterLowBalances],
+  );
+
```
Comment (truemiller) [inline frontend/hooks/useBalanceContext.ts:95]:
```suggestion
// use flatAddresses for consistency
```

File: frontend/hooks/useBalanceContext.ts (hunk 1 above)
Comment (truemiller) [reply frontend/hooks/useBalanceContext.ts:48]: will follow up with fixes, want to get this through so you can start working with the exports, assuming they are correct

File: frontend/hooks/useNeedsFunds.ts (hunk 4 above)
Comment (truemiller) [reply frontend/hooks/useNeedsFunds.ts:76]: yeah, i agree, however, boolean hides that the value is not yet loaded..
i.e. if it's true/false we can't assume that a new value is coming, or it is not yet defined

File: frontend/hooks/useBalanceContext.ts (hunk 2 above)
Comment (truemiller) [reply frontend/hooks/useBalanceContext.ts]: yeah, for sure, will update and add more consistently here

**Pull request merged**